source/defclass-star.lisp

```diff
@@ -522,3 +522,58 @@ See `define-class' for more details."
      `(define-condition ,name ,supers
         ,processed-slots
         ,@clean-options))))
+
+(defmacro make-instance* (class &rest rest)
+  "Convenience macro on top of `make-instance'.
+
+Conveniences:
+- First argument can be a list of symbols, which is automatically
+  converted into a keyword+symbol pairs.
+- The last argument can be an arbitrary list-producing form, which
+  will be used as the last argument to (apply #'make-instance ...).
+
+Otherwise, the behavior is equivalent to `make-instance'.
+
+Examples:
+
+Shortcut args:
+\(make-instance* 'class (a b c))
+=>
+\(make-instance 'class :a a :b b :c c)
+
+Last form for `apply':
+\(make-instance* 'class :a a :b b (when something (list a)))
+=>
+\(apply #'make-instance 'class :a a :b b (when something (list a)))
+
+Shortcut arguments, regular arguments, and form for `apply':
+\(make-instance* 'class (a b c) :x x (when something (list a)))
+=>
+\(apply #'make-instance 'class :a a :b b :c c :x x (when something (list a)))"
+  (let* ((maybe-shortcut-arguments (first rest))
+         (shortcuts-p (and (listp maybe-shortcut-arguments)
+                           (every #'symbolp maybe-shortcut-arguments)
+                           (not (fboundp (first maybe-shortcut-arguments)))))
```

Comment:
This condition is brittle.  If the first arg is fbound and also the slot-name, then `make-instance*` won't  be usable.
I suggest removing it.  After all, if the user can always use `make-instance` if they need to call a function.

Reply:
Hmmmm, maybe I'll need to require the shortcuts arg after all...